File: android/src/main/java/au/com/crazybean/mobilex/kurir/modules/explore/ExploreModule.kt
package au.com.crazybean.mobilex.kurir.modules.explore

import au.com.crazybean.mobilex.kurir.extension.qualifier
import au.com.crazybean.mobilex.kurir.modules.base.ViewModel
import org.koin.androidx.viewmodel.dsl.viewModel
import org.koin.dsl.module

val exploreModule = module {
    val qualifier = qualifier<ExploreWrapper>()

    // Wrapper via ViewModel
    viewModel(qualifier) {
        ViewModel(ExploreWrapper(get()))
    }

    // Actor
    factory { (scene: ExploreScene?) ->
        val viewModel = get<ViewModel<ExploreWrapper>>(qualifier)
        ExploreActor(scene, viewModel.wrapper)
    }
}

File: ios/Kurir/modules/explore/ExploreModule.swift
//
//  ExploreModule.swift
//  Kurir
//
//  Created by Loren on 19/9/19.
//  Copyright © 2019 Crazybean Studio. All rights reserved.
//

import Mobilex

class ExploreModule: Module {
    override func inject() {
        // Wrapper
        factory(ExploreWrapper.self) { r in
            ExploreWrapper(repository: r.resolve(UsersRepository.self))
        }
        
        // Actor
        factory(ExploreActor.self) { (r, scene: ExploreScene) -> ExploreActor in
            ExploreActor(scene: scene, wrapper: r.resolve(ExploreWrapper.self)!)
        }
    }
}
